Use m3u files for playlists

kraetzin opened the pull request:
The aim here is to switch playlists over from using a single pickled file to using multiple files in the m3u format (see #112).

I have implemented this by creating a folder called `playlists` in the mpsyt directory which expects playlists in the form `<playlist_name>.m3u`.  All existing mpsyt playlists get exported from `playlist_v2` into this directory. The extended `#EXTM3U` format is used whenever mpsyt saves playlists, but it can also handle a simple list of urls. Each m3u file is opened and checked for the header `#EXTM3U`, and if it does not contain this on the first line then a list of urls (or YTIDs) is expected.

This currently needs testing and I'm certain there's many unhandled errors, especially with regards to older mpsyt playlist formats. I'm also planning to switch the `play_history` file to m3u (effectively implements #667 and #681), but would like to make sure this is a sane implementation first.

I don't consider this finished and would love feedback on the changes :smiley:.

ids1024 commented:
This looks good in general. It's hard to check every edge case, but hopefully it can't break anything permanently since the old playlist file is kept.

One small issue: deleting a playlist with `rmp` needs to be handled. With the current implementation, it doesn't remove the file, so the playlist is still there when mpsyt is opened again.

kraetzin commented:
I believe I've fixed the `rmp` issue. `mv` also works as expected now. I've moved some exceptions from `load()`, that don't apply to the new format, to the conversion function, but I wouldn't know how to test those. I could remove them completely but I figured they do no harm and might help in some edge case.

kraetzin commented:
I've implemented the history of played songs as m3u and the conversion seems to be working fine. The new file is simply called `play_history.m3u`.
